### DISCUSSION TITLE
Bold fonts ? (on Mac)

### DISCUSSION THREAD

**Discussion by grege2** [eteran/nedit-ng#237] [General]
Ciao tutti, buon Capodanno, e felice nuovo anno in generale.
(Scusate, sto imparando l'italiano).

It could be useful at times to use a bold font, for better visibility on screen, or better printing. My Mac is not offering me any bold fonts. In fact, for your interest, it's offering me a strange set of fonts, but they appear to be the set of fixed-width fonts the Mac has. 

From nedit-ng ..

![nedit-ng-mac-fonts](https://user-images.githubusercontent.com/33416604/103434034-8dcbc080-4c4f-11eb-9f09-27f6d5e0318f.png)

From Mac TextEdit app .. there are bold versions for most of the fixed-width fonts:

![mac-textedit-fonts](https://user-images.githubusercontent.com/33416604/103434041-9cb27300-4c4f-11eb-81d3-f2851990aea4.png)

Anyway, any thoughts on accessing bold fonts on Mac, or any platform ?

Grazie.

**Comment by eteran** [comment]
Yes, NG only supports fixed-width fonts. That was a decision made for numerous reasons, chief among them is that GREATLY simplifies the painting algorithms, but also that for a code editor, a variable width font seems... well, I don't understand why someone would want it ;-).

As for bold, well NEdit does offer for **styles** to be bold. If you go to: Preferences -> Default Settings -> Syntax Highlighting -> Text Drawing Patterns. You can set the color, boldness, or italics of a given style. Just note that it applies to a specific style and not the font as a whole. This is because something like a keyword may want to be bold in comparison to the normal font.

Not exactly what you're asking for, I know. You may also want to look into some fonts that are in themselves, bolder (and when bolded are even bolder). A great choice IMO is "Source Code Pro" which offers "Medium" "Semibold" and "Black" variants which are degrees of being bolder than the normal variant.

https://fonts.google.com/specimen/Source+Code+Pro

Hack is also just generally a nice looking editor font :-)

https://github.com/source-foundry/Hack

**Comment by grege2** [comment]
Hi, thanks for all that. I'm not a great fonts expert on Mac, but your tips have led me to what I wanted. I had settled on Consolas in nedit-ng, and I've been using Inconsolata in another project, and a font comparison site says they're 99.9% identical 

http://www.identifont.com/differences?first=Consolas&second=Inconsolata&q=Go

So I downloaded Inconsolata ..   

https://fonts.google.com/specimen/Inconsolata?preview.size=79&preview.text=abcdefdefg&preview.text_type=custom&query=incons

and it was a bit condensed (narrower) than what I was used to, but SemiExpanded-Bold was just right.

Consolas ..

![Consolas](https://user-images.githubusercontent.com/33416604/103447411-1a718f80-4cdf-11eb-9dee-8d5587adaac0.png)

Inconsolata SemiExpanded-Bold ..

![Inconsolata](https://user-images.githubusercontent.com/33416604/103447417-2e1cf600-4cdf-11eb-80a9-dfa5c50c92eb.png)

Those images don't look great here, but on screen they're very clear. So I can flip between those two when I want more boldness. And there's still SemiExpanded-{ Medium, ExtraBold, Black } as more options.

Buonissimo ! Ciao.
